modules/javafx.controls/src/test/java/test/javafx/scene/control/TableViewTest.java

```diff
@@ -5848,4 +5850,87 @@ public void testAnchorRemainsWhenAddingMoreItemsBelow() {
         sl.dispose();
     }
 
+    // See JDK-8087673
+    @Test
+    public void testTableMenuButtonDoesNotOverlapColumnHeaderGraphic() {
+        TableView<String> table = new TableView<>();
+        table.setColumnResizePolicy(TableView.CONSTRAINED_RESIZE_POLICY);
+        table.setTableMenuButtonVisible(true);
+        TableColumn<String, String> column = new TableColumn<>();
+        Slider slider = new Slider();
+        slider.setValue(100);
+        column.setGraphic(slider);
+        table.getColumns().add(column);
+
+        StageLoader sl = new StageLoader(table);
+
+        Toolkit.getToolkit().firePulse();
+
+        ScrollBar vbar = VirtualFlowTestUtils.getVirtualFlowVerticalScrollbar(table);
+        assertFalse(vbar.isVisible());
+
+        StackPane thumb = (StackPane) slider.lookup(".thumb");
+        assertNotNull(thumb);
+        double thumbMaxX = thumb.localToScene(thumb.getLayoutBounds()).getMaxX();
+
+        StackPane corner = (StackPane) table.lookup(".show-hide-columns-button");
+        assertNotNull(corner);
+        assertTrue(corner.isVisible());
+        double cornerMinX = corner.localToScene(corner.getLayoutBounds()).getMinX();
+
+        assertTrue(thumbMaxX < cornerMinX);
+
+        sl.dispose();
```

Comment:
may be in try/finally to dispose of the StageLoader if test throws an exception?

Reply:
There are no exceptions expected to happen in these new tests and I don't see that we use try-finally with `StageLoader` by default, is this change really required?

Reply:
What we usually do is to add a global variable like this:
`private StageLoader stageLoader;`
and in the after method something like this:
`if (stageLoader != null) stageLoader.dispose();`

So tests are free to init and use the `stageLoader` and it will always call `dispose()` after the test, no matter if it failed or not.

cc @kleopatra 
see for example: `TableCellTest`

Reply:
Thanks @Maran23, that is useful.